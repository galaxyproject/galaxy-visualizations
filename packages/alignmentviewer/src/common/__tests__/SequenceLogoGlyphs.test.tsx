import "jest-webgl-canvas-mock";
import * as React from "react";

import { shallow, default as Enzyme } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter() });

import { GlyphFactory } from "../SequenceLogoGlyphs";

describe("SequenceLogoGlyphs", () => {
  it.each([
    "A",
    "B",
    "C",
    "D",
    "E",
    "F",
    "G",
    "H",
    "I",
    "J",
    "K",
    "L",
    "M",
    "N",
    "O",
    "P",
    "Q",
    "R",
    "S",
    "T",
    "U",
    "V",
    "W",
    "X",
    "Y",
    "Z"
  ])("Should render the correct glyph for '%s'", char => {
    const glyph = GlyphFactory.glyphFromChar(char)({
      className: `class-name-for-${char}`,
      key: `key-for-${char}`
    });
    const wrapper = shallow(glyph);
    expect(wrapper).toMatchSnapshot();
  });
});
