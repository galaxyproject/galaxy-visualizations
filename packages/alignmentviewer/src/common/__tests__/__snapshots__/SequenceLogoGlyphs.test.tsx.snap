// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SequenceLogoGlyphs Should render the correct glyph for 'A' 1`] = `
<g
  key="key-for-A_g"
>
  <path
    className="class-name-for-A"
    d="M 0 100 L 33 0 L 66 0 L 100 100 L 75 100 L 66 75 L 33 75 L 25 100 L 0 100"
    key="key-for-A"
  />
  <path
    d="M 41 55 L 50 25 L 58 55 L 41 55"
    fill="#ffffff"
    key="key-for-A_p2"
  />
</g>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'B' 1`] = `
<g
  key="key-for-B_g"
>
  <path
    className="class-name-for-B"
    d="M 0 0 L 80 0 C 105 0 105 50 80 50 C 105 50 105 100 80 100 L 00 100 L 0 0"
    key="key-for-B"
  />
  <path
    d="M 20 15 L 70 15 C 80 15 80 35 70 35 L 20 35 L 20 15"
    fill="#ffffff"
    key="key-for-B_p2"
  />
  <path
    d="M 20 65 L 70 65 C 80 65 80 85 70 85 L 20 85 L 20 65"
    fill="#ffffff"
    key="key-for-B_p3"
  />
</g>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'C' 1`] = `
<path
  className="class-name-for-C"
  d="M 100 28 C 100 -13 0 -13 0 50 C 0 113 100 113 100 72 L 75 72 C 75 90 30 90 30 50 C 30 10 75 10 75 28 L 100 28"
  key="key-for-C"
/>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'D' 1`] = `
<g
  key="key-for-D_g"
>
  <path
    className="class-name-for-D"
    d="M 0 0 L 60 0 C 110 0 110 100 60 100 L 0 100 L 0 0"
    key="key-for-D"
  />
  <path
    d="M 20 15 L 40 15 C 85 15 85 85 40 85 L 20 85 L 20 15"
    fill="#ffffff"
    key="key-for-D_p2"
  />
</g>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'E' 1`] = `
<path
  className="class-name-for-E"
  d="M 0 0 L 100 0 L 100 20 L 20 20 L 20 40 L 90 40 L 90 60 L 20 60 L 20 80 L 100 80 L 100 100 L 0 100 L 0 0"
  key="key-for-E"
/>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'F' 1`] = `
<path
  className="class-name-for-F"
  d="M 0 0 L 100 0 L 100 20 L 20 20 L 20 40 L 80 40 L 80 60 L 20 60 L 20 100 L 0 100 L 0 0"
  key="key-for-F"
/>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'G' 1`] = `
<path
  className="class-name-for-G"
  d="M 100 28 C 100 -13 0 -13 0 50 C 0 113 100 113 100 72 L 100 48 L 55 48 L 55 72 L 75 72 C 75 90 30 90 30 50 C 30 10 75 5 75 28 L 100 28"
  key="key-for-G"
/>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'H' 1`] = `
<path
  className="class-name-for-H"
  d="M 0 0 L 20 0 L 20 40 L 80 40 L 80 0 L 100 0 L 100 100 L 80 100 L 80 60 L 20 60 L 20 100 L 0 100 L 0 0"
  key="key-for-H"
/>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'I' 1`] = `
<path
  className="class-name-for-I"
  d="M 40 0 L 60 0 L 60 100 L 40 100 L 40 0"
  key="key-for-I"
/>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'J' 1`] = `
<path
  className="class-name-for-J"
  d="M 0 60 C 0 111 100 111 100 60 L 100 0 L 75 0 L 75 60 C 80 90 20 90 25 60"
  key="key-for-J"
/>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'K' 1`] = `
<path
  className="class-name-for-K"
  d="M 0 0 L 20 0 L 20 40 L 75 0 L 100 0 L 50 50 L 100 100 L 75 100 L 30 65 L 20 75 L 20 100 L 0 100 L 0 0"
  key="key-for-K"
/>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'L' 1`] = `
<path
  className="class-name-for-L"
  d="M 0 0 L 0 100 L 100 100 L 100 80 L 20 80 L 20 0 L 0 0"
  key="key-for-L"
/>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'M' 1`] = `
<path
  className="class-name-for-M"
  d="M 0 0 L 20 0 L 50 35 L 80 0 L 100 0 L 100 100 L 80 100 L 80 30 L 50 65 L 20 30 L 20 100 L 0 100 L 0 0"
  key="key-for-M"
/>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'N' 1`] = `
<path
  className="class-name-for-N"
  d="M 0 100 L 0 0 L 20 0 L 80 75 L 80 0 L 100 0 L 100 100 L 80 100 L 20 25 L 20 100 L 0 100"
  key="key-for-N"
/>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'O' 1`] = `
<g
  key="key-for-O_g"
>
  <circle
    className="class-name-for-O"
    cx="50"
    cy="50"
    key="key-for-O"
    r="50"
  />
  <circle
    cx="50"
    cy="50"
    fill="#ffffff"
    key="key-for-O_c2"
    r="32"
  />
</g>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'P' 1`] = `
<g
  key="key-for-P_g"
>
  <path
    className="class-name-for-P"
    d="M 0 0 L 80 0 C 105 0 105 50 80 50 L 20 50 L 20 100 L 0 100 L 0 0"
    key="key-for-P"
  />
  <path
    d="M 20 15 L 70 15 C 80 15 80 35 70 35 L 20 35 L 20 15"
    fill="#ffffff"
    key="key-for-P_p2"
  />
</g>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'Q' 1`] = `
<g
  key="key-for-Q_g"
>
  <circle
    className="class-name-for-Q"
    cx="50"
    cy="50"
    key="key-for-Q_c1"
    r="50"
  />
  <circle
    cx="50"
    cy="50"
    fill="#ffffff"
    key="key-for-Q_c2"
    r="32"
  />
  <path
    className="class-name-for-Q"
    d="M 85 100 L 55 70 L 70 55 L 100 85 L 85 100"
    key="key-for-Q"
  />
</g>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'R' 1`] = `
<g
  key="key-for-R_g"
>
  <path
    className="class-name-for-R"
    d="M 0 0 L 80 0 C 105 0 105 50 80 50 C 100 50 100 70 100 70 L 100 100 L 80 100 L 80 80 C 80 80 80 60 50 60 L 20 60 L 20 100 L 0 100 L 0 0"
    key="key-for-R"
  />
  <path
    d="M 20 15 L 70 15 C 80 15 80 35 70 35 L 20 35 L 20 15"
    fill="#ffffff"
    key="key-for-R_p2"
  />
</g>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'S' 1`] = `
<path
  className="class-name-for-S"
  d="M92 26 A43 20 0 1 0 43 46 A42 23 0 1 1 9 68"
  fill="#ffffff"
  key="key-for-S"
  strokeWidth="18"
/>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'T' 1`] = `
<path
  className="class-name-for-T"
  d="M 0 0 L 0 20 L 35 20 L 35 100 L 65 100 L 65 20 L 100 20 L 100 0 L 0 0"
  key="key-for-T"
/>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'U' 1`] = `
<path
  className="class-name-for-U"
  d="M 0 0 L 0 60 C 0 111 100 111 100 60 L 100 0 L 75 0 L 75 60 C 80 90 20 90 25 60 L 25 0 L 0 0"
  key="key-for-U"
/>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'V' 1`] = `
<path
  className="class-name-for-V"
  d="M 0 0 L 20 0 L 50 80 L 80 0 L 100 0 L 60 100 L 40 100 L 0 0"
  key="key-for-V"
/>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'W' 1`] = `
<path
  className="class-name-for-W"
  d="M 0 0 L 20 0 L 30 70 L 50 30 L 70 70 L 80 0 L 100 0 L 90 100 L 70 100 L 50 65 L 30 100 L 10 100 L 0 0"
  key="key-for-W"
/>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'X' 1`] = `
<path
  className="class-name-for-X"
  d="M 0 0 L 20 0 L 50 40 L 80 0 L 100 0 L 70 50 L 100 100 L 80 100 L 50 60 L 20 100 L 0 100 L 30 50 L 0 0"
  key="key-for-X"
/>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'Y' 1`] = `
<path
  className="class-name-for-Y"
  d="M 0 0 L 20 0 L 50 45 L 80 0 L 100 0 L 60 60 L 60 100 L 40 100 L 40 60 L 0 0"
  key="key-for-Y"
/>
`;

exports[`SequenceLogoGlyphs Should render the correct glyph for 'Z' 1`] = `
<path
  className="class-name-for-Z"
  d="M 0 0 L 100 0 L 100 20 L 35 80 L 100 80 L 100 100 L 0 100 L 0 80 L 65 20 L 0 20 L 0 0"
  key="key-for-Z"
/>
`;
